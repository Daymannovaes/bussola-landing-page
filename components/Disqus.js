import React from 'react';
import { DiscussionEmbed } from 'disqus-react';
import PropTypes from 'prop-types';

import { DISQUS_SHORTNAME, SITE_URL } from '../config/globals';

const Disqus = ({ id, title }) => (
    <div>
        <h4>Coment√°rios</h4>

        <DiscussionEmbed
            shortname={DISQUS_SHORTNAME}
            config={{
                title,
                identifier: id,
                url: `${SITE_URL}/${id}`,
            }}
        />
    </div>
);

Disqus.propTypes = {
    title: PropTypes.string.isRequired,
    id: PropTypes.string.isRequired,
};

export default Disqus;
