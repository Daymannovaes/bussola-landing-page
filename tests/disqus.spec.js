import React from 'react';
import Disqus from '../components/Disqus';
import renderer from 'react-test-renderer';

describe('disqus.spec.js', () => {
    it('snapchot', () => {
        const component = renderer.create(
            <Disqus id="some-id" title="some title" />
        );

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

});
