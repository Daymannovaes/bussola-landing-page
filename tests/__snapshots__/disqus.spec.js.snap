// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`disqus.spec.js snapchot 1`] = `
<div>
  <h4>
    Coment√°rios
  </h4>
  <div
    id="disqus_thread"
  />
</div>
`;
